main/tests/test_os.cpp

```diff
@@ -0,0 +1,75 @@
+/*************************************************************************/
+/*  test_os.cpp                                                    */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2020 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2020 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "test_os.h"
+
+#include "core/os/main_loop.h"
+#include "servers/display_server.h"
+#include "servers/rendering_server.h"
+
+namespace TestOS {
+
+bool test_get_latin_keyboard_variant() {
+	print_line("test_get_latin_keyboard_variant");
+	const DisplayServer::LatinKeyboardVariant r = DisplayServer::get_singleton()->get_latin_keyboard_variant();
+	OS::get_singleton()->print("latin_keyboard_variant: %d\n", r);
+	// As today get_latin_keyboard_variant cannot really fails as it returns by default QWERTY in case of any failure. That test at least checks for any crash.
+	return true;
+};
```

Comment:
To be fair I don't see much point to add an `OS` test if it's testing a single function which can't fail.

There's definitely need for a better unit testing coverage for the core API, but we've been discussing about using a proper framework to do this (see https://github.com/godotengine/godot-proposals/issues/918) so until this is properly integrated, adding such tests that nobody will run is not very useful.